__tests__/routes/note.test.js

```diff
@@ -0,0 +1,371 @@
+    expect(process.env.NODE_ENV).toBe('test');
+  });
+});
+
+describe('Notes Router', () => {
+  describe('[GET] /notes', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes');
+      });
+
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+    });
+    describe('failed', () => {});
+  });
+
+  describe('[GET] /notes/note_id', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes/1');
+      });
+
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+
+      it('responds with non empty object', async () => {
+        const expected = /some text here/i;
+        const actual = res.text;
+        expect(actual).toMatch(expected);
+      });
+    });
+    describe('failed', () => {
+      describe('case - failed 404 error', () => {
+        let res;
+        beforeAll(async () => {
+          res = await request(app).get('/notes/100');
+        });
+
+        it('requires authentication', () => {
+          expect(authRequired).toBeCalled();
+        });
+
+        it('responds with status 200', async () => {
+          const expected = 404;
+          const actual = res.status;
+          expect(actual).toBe(expected);
+        });
+      });
+    });
+  });
+
+  describe('[GET] /notes/mentee/mentee_id', () => {
+    describe('succeed', () => {
+      let res;
+      beforeAll(async () => {
+        res = await request(app).get('/notes/mentee/00ultwqjtqt4VCcS24x6');
+      });
+      it('requires authentication', () => {
+        expect(authRequired).toBeCalled();
+      });
+      it('responds with status 200', async () => {
+        const expected = 200;
+        const actual = res.status;
+        expect(actual).toBe(expected);
+      });
+
+      it('responds with non empty object', async () => {
+        const expected = 5;
+        const actual = res.body.length;
+        expect(actual).toBe(expected);
+      });
+    });
+
+    describe('failed', () => {
+      describe('case - failed 404 error', () => {
+        let res;
+        beforeAll(async () => {
+          res = await request(app).get('/notes/mentee/-1');
+        });
+        it('requires authentication', () => {
+          expect(authRequired).toBeCalled();
+        });
+        it('responds with status 200', async () => {
+          const expected = 404;
+          const actual = res.status;
+          expect(actual).toBe(expected);
+        });
+      });
+
+      it('sanity', () => {
```

Comment:
I don't love this test. Sanity tests aren't always my favorite, but being that testing stays in Development I'm ok with them.